samples/base/Logger.cpp

```diff
@@ -60,9 +64,13 @@ namespace darwin {
             fmt << "\"filter\":\"" << _name << "\",";
             fmt << "\"message\":\"" << logMsg << "\"";
             fmt << '}';
-            _fileMutex.lock();
-            _file << fmt.str().c_str() << std::endl;
-            _fileMutex.unlock();
+            if (not _file.is_open() and not this->openLogFile()) {
+                std::clog << fmt.str() << std::endl;
+            } else {
+                _fileMutex.lock();
```

Comment:
best practice to use RAII with `std::lock_guard lock(_fileMutex)` and let it be destroyed by the end of the scope

Reply:
good call

Reply:
change made, I leave you to comment/resolve the conversation